readthedocs/projects/views/private.py

```diff
@@ -955,6 +971,80 @@ def get_success_url(self):
         return reverse('projects_integrations', args=[self.get_project().slug])
 
 
+class IntegrationConnectProject(IntegrationMixin, GenericView):
+
+    """
+    Connect a ``Project`` to a ``RemoteRepository``.
+
+    Useful when the Project has been imported manually or got disconnected from
+    the ``RemoteRepository`` instance for some reason.
+
+    Connecting the repository will allow people to handle permissions access
+    permissions (VCS SSO) and also when sending build status notifications,
+    among others.
+    """
+
+    @staticmethod
+    def get_final_url(project):
```

Comment:
we should put a docstring about this. The head request looks like it's to get normalize `http` and `.git` urls?